eth_utils/address.py

```diff
@@ -79,15 +84,15 @@ def is_normalized_address(value):
         return value == to_normalized_address(value)
 
 
-def to_canonical_address(address):
+def to_canonical_address(address: Address) -> bytes:
```

Comment:
The input here should probably be [`AnyStr`](https://docs.python.org/3/library/typing.html#typing.AnyStr) and the output should be whatever `NewType` you end up using for canonical addresses (bytes).